subsys/bluetooth/services/dis.c

```diff
@@ -0,0 +1,92 @@
+/** @file
+ *  @brief BLE GATT Device Infromation Service
+ */
+
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <init.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#define DIS_PNP_ID_LEN	7	/** Length of PnP ID Characteristic Value */
+
+static const char dis_model[] = CONFIG_BT_GATT_DIS_MODEL;
```

Comment:
You can pass the config options when declaring the characteristic, then access it with attr->user_data, that way we avoid having static pointers.

Reply:
Done